python/shotgun_fields/shotgun_field_delegate.py

```diff
@@ -12,16 +12,30 @@
 from sgtk.platform.qt import QtCore, QtGui
 
 views = sgtk.platform.current_bundle().import_module("views")
-shotgun_globals = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_globals")
-shotgun_model = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_model")
 
-class ShotgunFieldDelegate(views.WidgetDelegate):
+shotgun_globals = sgtk.platform.import_framework(
+    "tk-framework-shotgunutils", "shotgun_globals")
+
+shotgun_model = sgtk.platform.import_framework(
+    "tk-framework-shotgunutils", "shotgun_model")
+
+
+class ShotgunFieldDelegateGeneric(views.WidgetDelegate):
     """
-    A delegate for a given type of Shotgun field.  This delegate is designed to work
-    with indexes from a ShotgunModel where the value of the field is stored in the
-    SG_ASSOCIATED_FIELD_ROLE role.
+    A generic, model-agnostic, shotgun field widget delegate.
+
+    This class is designed to be used with any model that represents data that
+    can be stored in Shotgun fields.
+
+    The included ``ShotgunFieldDelegate`` class is designed to work specifically
```

Comment:
Yeah, I thought about that. It does feel a little weird, but since they're in the same file I wanted to make sure anyone reading the docs for the base class knew about the subclass as well in case they were looking for the SG model delegate.